src/controller/p2p/preheat/controller.go

```diff
@@ -197,14 +207,38 @@ func (c *controller) CountPolicy(ctx context.Context, query *q.Query) (int64, er
 	return c.pManager.Count(ctx, query)
 }
 
+// TriggerParam ...
+type TriggerParam struct {
+	PolicyID int64
+}
+
 // CreatePolicy creates the policy.
-func (c *controller) CreatePolicy(ctx context.Context, schema *policyModels.Schema) (int64, error) {
+func (c *controller) CreatePolicy(ctx context.Context, schema *policyModels.Schema) (id int64, err error) {
 	if schema != nil {
 		now := time.Now()
 		schema.CreatedAt = now
 		schema.UpdatedTime = now
 	}
-	return c.pManager.Create(ctx, schema)
+
+	if err = orm.WithTransaction(func(ctx context.Context) error {
+		if schema.Trigger != nil && schema.Trigger.Type == policyModels.TriggerTypeScheduled {
+			var cron = schema.Trigger.Settings.Cron
+			if len(cron) > 0 {
+				jobid, err := c.scheduler.Schedule(ctx, cron, SchedulerCallback, TriggerParam{PolicyID: schema.ID})
```

Comment:
It seems at this time there is no policy ID yet because it is not persisted yet

Reply:
fixed.